src/coreclr/tools/Common/Internal/Metadata/NativeFormat/NativeFormatReaderGen.cs

```diff
@@ -95,17 +95,17 @@ public override bool Equals(object obj)
         public static implicit operator Handle(ArraySignatureHandle handle)
             => new Handle(handle._value);
 
-        internal int Offset => (_value & 0x00FFFFFF);
+        internal int Offset => (_value & 0x01FFFFFF);
 
         public ArraySignature GetArraySignature(MetadataReader reader)
             => new ArraySignature(reader, this);
 
-        public bool IsNil => (_value & 0x00FFFFFF) == 0;
+        public bool IsNil => (_value & 0x01FFFFFF) == 0;
 
         [System.Diagnostics.Conditional("DEBUG")]
         internal void _Validate()
         {
-            if ((HandleType)((_value & 0xFF000000) >> 24) != HandleType.ArraySignature)
+            if ((HandleType)((uint)(_value & 0xFF000000) >> 25) != HandleType.ArraySignature)
```

Comment:
```suggestion
            if ((HandleType)((uint)value >> 25) != HandleType.ArraySignature)
```
This masking is not needed and it does not make sense now that the shift is 25.

Reply:
Ah, good catch, yes, it doesn't make sense.